src/main/java/servlets/UploadReceiptServlet.java

```diff
@@ -181,6 +204,7 @@ private Entity analyzeReceiptImage(BlobKey blobKey, HttpServletRequest request)
         results = ReceiptAnalysis.serveImageText(absoluteUrl);
       }
     } catch (IOException e) {
+      blobstoreService.delete(blobKey);
```

Comment:
Also this too, for future reference, I would include in a different pull request.